lib/srv/usermgmt.go

```diff
@@ -0,0 +1,220 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package srv
+
+import (
+	"errors"
+	"io"
+	"os/user"
+	"runtime"
+	"time"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/trace"
+	"github.com/siddontang/go/log"
+)
+
+// NewHostUsers initialize a new HostUsers object
+func NewHostUsers() (HostUsers, error) {
+	backend, err := newHostUsersBackend()
+	if err != nil {
+		return nil, err
+	}
+	return &HostUserManagment{
+		backend:       backend,
+		cancelCleanup: make(chan bool),
+	}, nil
+}
+
+// newHostUsersBackend initializes a new OS specific HostUsersBackend
+func newHostUsersBackend() (HostUsersBackend, error) {
+	if runtime.GOOS == "windows" {
+		return nil, trace.NotImplemented("Host user creation management is only supported on linux")
+	}
+	return &UnixHostUsersBackend{}, nil
```

Comment:
`runtime.GOOS` is defined at run time, so this will actually fail to compile on windows.

Reply:
I've changed this so there is now an implementation on linux and an implementation on everything else with dummy methods so i think compilation should continue to work when compiling on other platforms now